step2-04/README.md

```diff
@@ -1,28 +1,27 @@
-# Step 2.4
+# Step 2.4: Testing TypeScript code with Jest
 
 [Lessons](../) | [Exercise](./exercise/) | [Demo](./demo/)
 
-Testing TypeScript code with jest. jest is a test framework made by Facebook and is very popular in the React and the wider JS ecosystem. We will work on implementing simple unit tests here in this exercise.
+[Jest](https://jestjs.io/) is a test framework made by Facebook and is very popular in the React and wider JS ecosystems.
 
-https://jestjs.io/
+In this exercise, we will work on implementing simple unit tests using Jest.
 
-# jest Features
+## Jest Features
 
 - Multi-threaded and isolated test runner
-- Provides a "fake" browser environment if needed (window, document, DOM, etc).
-- Snapshots: show API or large object changes along side code changes in pull requests
-- Code coverage is integrated (--coverage)
-- Very clear error messages of where the test failures occur
-- By default, will simulate a "good enough" browser environment called JSDOM
+- Provides a fake browser-like environment if needed (window, document, DOM, etc) using jsdom
+- Snapshots: Jest can create text-based snapshots of rendered components. These snapshots can be checked in and show API or large object changes alongside code changes in pull requests.
+- Code coverage is integrated (`--coverage`)
+- Very clear error messages showing where a test failure occurred
 
-# How to use jest
+## How to use Jest
 
-- using `create-react-app` or other project generators, jest should already be preconfigured. Run `npm test` usually will trigger it!
-- needs `jest.config.js`
+- Using `create-react-app` or other project generators, Jest should already be pre-configured. Running `npm test` usually will trigger it!
```

Comment:
This still feels more like presenter notes (I had a harder time converting it to text than with some other sections)